### DISCUSSION TITLE
appsettings.json configuration

### DISCUSSION THREAD

**Discussion by lucadecamillis** [dadhi/DryIoc#590] [Ideas]
Hi,

Let me first thank you for your nice work here :-)

I would like to have the option to set up container registrations via configuration file (a.k.a. appsettings.json) like Unity does via xml configuration. I'm not sure whether this feature is already implemented in the library (I haven't found it yet).

I'm of course available for the development of the feature and I wonder whether you might consider to add the feature to the library. Otherwise I'll just write an external library to do the job and I'll deploy it separately.

Just let me know what you think.

Thanks

**Comment by dadhi** [comment]
@lucadecamillis 
Hi, I did not include any separate configuration because it may very much differ based on your context.
And to be fully transparent, I am a proponent of configuration as code.

I don't know your situation, but I think you may build it quite easily. 
Appsettings.json has the standard Configuration API, so you may load info about your registrations and convent into the actual code.
If you don't need the runtime updates, you may even do this in compile-time with Source Generators...

Likely, someone already did that, :) So you may try to find it and adopt for DryIoc.

**Reply by lucadecamillis** [reply]
What I need is to have the ability to shape the behavior of a certain executable based on the container configuration. So at runtime the choice of the registered components can alter the flavor of the application (the application is question is a WPF PRISM executable).

I would indeed bind some registration configuration via the Configuration API and based on that I would dynamically populate the container based on the configuration. To have an idea of what I want to achieve, I've kind of done the same for the Microsoft DependencyInjection API (https://github.com/lucadecamillis/dependency-injection-configuration)

In this particular situation source generators may not be suitable to fullfil the requirement above described.

**Reply by dadhi** [reply]
Great, 
So you may use DryIoc.Microsoft.DeendencyInjection package to pass the service collection into DryIoc Container.
Or you want the specific DryIoc registration features?

**Reply by lucadecamillis** [reply]
@dadhi
I had not planned to include the MS.DependencyInjection package in the project but at first sight it might be just enough to achive my goal.

It can be that along the way other container features will be necessary, at that point I'll reconsider the idea.

In the meanwhile thanks for the tip!
